`dotnet test -f net8.0-windows` fails with `System.NullReferenceException`

[My CI #1839](https://github.com/y-iihoshi/ThScoreFileConverter/actions/runs/7151270748) failed (at Attempt 1 through 4) with the following error:

```
Run dotnet test ThScoreFileConverter.Core.Tests --configuration Debug --framework net8.0-windows -e CollectCoverage=true -e CoverletOutput="../CoverageResults/"
  dotnet test ThScoreFileConverter.Core.Tests --configuration Debug --framework net8.0-windows -e CollectCoverage=true -e CoverletOutput="../CoverageResults/"
  dotnet test ThScoreFileConverter.Tests --configuration Debug --framework net8.0-windows -e CollectCoverage=true -e CoverletOutput="../CoverageResults/" -e MergeWith="../CoverageResults/coverage.net8.0-windows.json" -e CoverletOutputFormat=opencover

...omitted...

Test run for D:\a\ThScoreFileConverter\ThScoreFileConverter\ThScoreFileConverter.Tests\bin\Debug\net8.0-windows\ThScoreFileConverter.Tests.dll (.NETCoreApp,Version=v8.0)
Microsoft (R) Test Execution Command Line Tool Version 17.8.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
The active test run was aborted. Reason: Test host process crashed : Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Threading.Thread.get_CurrentThread()
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)


Test Run Aborted.
Passed!  - Failed:     0, Passed:  4111, Skipped:     0, Total:  4111, Duration: 19 s - ThScoreFileConverter.Tests.dll (net8.0)
Error: Process completed with exit code 1.
```

--- a/ThScoreFileConverter.Tests/UnitTesting/STATestMethodAttribute.cs
+++ b/ThScoreFileConverter.Tests/UnitTesting/STATestMethodAttribute.cs
@@ -6,6 +6,18 @@ public sealed class STATestMethodAttribute : TestMethodAttribute
 {
     public override TestResult[] Execute(ITestMethod testMethod)
     {
+        // HACK: avoid null dereferences in net8.0 tests
+        //
+        // Although the root cause is unknown, accessing Thread.CurrentThread
+        // during testing can cause a NullReferenceException on net8.0.
+        // This behavior has only been observed on the GitHub-hosted runners
+        // (at least windows-2022 20231205.1.0 and 20231211.1.0).
+
+#if NET8_0_OR_GREATER
+        return [
+            new TestResult { Outcome = UnitTestOutcome.Inconclusive }
+        ];
+#else
         if (Thread.CurrentThread.GetApartmentState() == ApartmentState.STA)
             return base.Execute(testMethod);
 
@@ -15,5 +27,6 @@ public sealed class STATestMethodAttribute : TestMethodAttribute
         thread.Start();
         thread.Join();
         return result!;
+#endif
     }
 }

